client/app/vps/vps.js

```diff
@@ -1,117 +1,74 @@
-          templateUrl: 'app/vps/secondary-dns/vps-secondary-dns.html',
-          controller: 'VpsSecondaryDnsCtrl',
-          controllerAs: '$ctrl',
-        },
-      },
-    })
-    .state('iaas.vps.detail.cloud-database', {
-      url: '/cloud-database',
-      redirectTo: 'iaas.vps.detail.cloud-database.list',
-      views: {
-        vpsContent: {
-          template: '<div ui-view="vpsCloudDatabaseContent"></div>',
-        },
-      },
-      resolve: {
-        shouldNotBeCA: [
-          '$q',
-          'CucFeatureAvailabilityService',
-          ($q, CucFeatureAvailabilityService) => CucFeatureAvailabilityService
-            .hasFeaturePromise('VPS', 'cloudDatabase')
-            .then(hasFeature => (hasFeature ? $q.when() : $q.reject('Not authorized'))),
-        ],
-      },
-    })
-    .state('iaas.vps.detail.cloud-database.list', {
-      url: '/',
-      views: {
-        vpsCloudDatabaseContent: {
-          templateUrl: 'app/vps/cloud-database/vps-cloud-database.html',
-          controller: 'VpsCloudDatabaseCtrl',
-          controllerAs: '$ctrl',
+const moduleName = 'vpsModule';
+
+angular
+  .module(moduleName, [
+    vpsAdditionnalDiskModule,
+    vpsBackupStorageModule,
+    vpsCloudDatabaseModule,
+    vpsDashboardModule,
+    vpsMonitoringModule,
+    vpsSecondaryDnsModule,
+    vpsSnapshotModule,
+    vpsUpgradeModule,
+    vpsVeeamModule,
+    vpsWindowsModule,
+  ])
+  .component(detailComponent.name, detailComponent)
+  .component(headerComponent.name, headerComponent)
+  .config(/* @ngInject */($stateProvider) => {
+    $stateProvider
+      .state('iaas.vps', {
+        url: '/vps',
+        templateUrl: 'app/vps/vps.html',
+        abstract: true,
+        translations: {
+          value: ['../common', '../vps'],
+          format: 'json',
         },
-      },
-    })
-    .state('iaas.vps.detail.cloud-database.order', {
-      url: '/order',
-      views: {
-        vpsCloudDatabaseContent: {
-          templateUrl: 'app/vps/cloud-database/order/vps-cloud-database-order.html',
-          controller: 'VpsCloudDatabaseOrderCtrl',
-          controllerAs: '$ctrl',
+      })
+      .state('iaas.vps.detail', {
+        url: '/{serviceName}',
+        redirectTo: 'iaas.vps.detail.dashboard',
+        resolve: {
+          stateVps: /* @ngInject */ ($transition$, $q, OvhApiVps) => OvhApiVps.v6().get({
+            serviceName: _.get($transition$.params(), 'serviceName'),
+          }).$promise
+            .then(stateVps => OvhApiVps.v6().version({
+              serviceName: _.get($transition$.params(), 'serviceName'),
+            }).$promise.then((response) => {
+              _.set(stateVps, 'isFullAgora', response.version === 2);
+              return stateVps;
+            }))
+            .catch((error) => {
+              if (error.status === 404) {
+                return $q.reject(_.merge({ code: 'LOADING_STATE_ERROR' }, error));
+              }
+              return true;
+            }),
```

Comment:
We should try to avoid having module declaring several states imo :)

Reply:
Hmm the first state is just a abstract parent state, plus a solution could be to merge the two states, and so this requires to adapt one manager call to VPS section.

I agree with you, but this will lead to change outside of this repository, and so, the context of this PR.